### ISSUE TITLE
Infer nullability in switch on conditional access operator

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#34246]
A warning is reported for `s.ToString()` currently:
```C#
class Program
{
    static void M(string? s)
    {
        switch (s?.Length)
        {
            case 0:
                s.ToString();
                break;
        }
    }
}
```
